### PR TITLE
feat: add integration tests

### PR DESCRIPTION
**Pull request by lulf**
* Create integration-tests module where test infrastructure and tests live.
* Add integration test for bombastic that tests API, indexing and search.
* Ensure indexer starts from earliest offset to catch previous changes.

### CODE REVIEW

File: integration-tests/src/minio.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+/usr/bin/mc event add myminio/bombastic arn:minio:sqs::BOMBASTIC:kafka --event "put,delete";
+/usr/bin/mc mb myminio/vexination || true;
+/usr/bin/mc policy download myminio/vexination;
+/usr/bin/mc event add myminio/vexination arn:minio:sqs::VEXINATION:kafka --event put;
+/usr/bin/mc admin service restart myminio;"###;
+        Self {
+            args: vec!["-c".into(), COMMANDS.to_string()],
+        }
+    }
+}
```
Comment (jcrossley3) [inline integration-tests/src/minio.rs:111]: Again, this seems counter to relying on the compose.yaml file, which also has these commands.

File: integration-tests/tests/bombastic.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    run_test(Duration::from_secs(60), async move {
+        let client = reqwest::Client::new();
+        let input = serde_json::from_str::<Map<String, Value>>(include_str!("../../bombastic/testdata/ubi9-sbom.json"))
+            .unwrap();
+        let response = client
+            .post("http://localhost:8082/api/v1/sbom?id=ubi9")
+            .json(&input)
+            .send()
+            .await?;
+        assert_eq!(response.status(), StatusCode::CREATED);
```
Comment (jcrossley3) [inline integration-tests/tests/bombastic.rs:17]: So much code required just to run this, sheesh.

File: Cargo.toml [hunk 3]
```diff
@@ -24,9 +24,11 @@ members = [
...
     "index",
     "infrastructure",
+    "integration-tests",
 ]
 
 [patch.crates-io]
 #sikula = { git = "https://github.com/ctron/sikula", rev = "cea7429f6b690adcdcd1ce2edb2b9e2277834e03" }
 cyclonedx-bom = { git = "https://github.com/lulf/cyclonedx-rust-cargo", branch = "trustification" }
 spdx-rs = { git = "https://github.com/lulf/spdx-rs", rev = "f3f1ea9a72e270c8434f8be6ca1d61608b25b813" }
+bollard-stubs = { git = "https://github.com/lulf/bollard", branch = "bollard-podman" }
```
[the remark this answers is not in the document]
Comment (lulf) [reply Cargo.toml]: Not a norwegian word for sure :D

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -53,7 +53,7 @@ jobs:
           cargo check --manifest-path spog/ui/Cargo.toml
 
       - name: Test
-        run: cargo test
+        run: cargo test --workspace --exclude integration-tests
```
Comment (jcrossley3) [inline .github/workflows/ci.yaml:56]: I wouldn't expect this to be necessary

**Review by jcrossley3 — APPROVED** [review]
I dig it, as long as you're ok using `default-members`. Thanks for getting it done!

**Pull request merged**